src/System.Runtime.Analyzers/Core/SpecifyCultureInfo.cs

```diff
@@ -24,15 +27,86 @@ public abstract class SpecifyCultureInfoAnalyzer : DiagnosticAnalyzer
                                                                              s_localizableMessage,
                                                                              DiagnosticCategory.Globalization,
                                                                              DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
+                                                                             isEnabledByDefault: true,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: Uri,
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
+            analysisContext.RegisterCompilationStartAction(csaContext =>
+            {
+                var cultureInfoType = csaContext.Compilation.GetTypeByMetadataName("System.Globalization.CultureInfo");
+                if (cultureInfoType != null)
+                {
+                    csaContext.RegisterOperationAction(oaContext =>
+                    {
+                        var invocationExpression = (IInvocationExpression)oaContext.Operation;
+                        var targetMethod = invocationExpression.TargetMethod;
+                        if (targetMethod.ContainingType == null)
+                        {
+                            return;
+                        }
+
+                        var methodsWithSameNameAsTargetMethod = targetMethod.ContainingType.GetMembers(targetMethod.Name).OfType<IMethodSymbol>();
+                        if (methodsWithSameNameAsTargetMethod.Count() < 2)
+                        {
+                            return;
+                        }
+
+                        var correctOverloads = methodsWithSameNameAsTargetMethod.Where(candidateMethod =>
+                        {
+                            if (targetMethod.Parameters.Count() + 1 != candidateMethod.Parameters.Count())
+                            {
+                                return false;
+                            }
+
+                            // Assume CultureInfo is the last parameter so j, which is the index of the parameter
```

Comment:
The expected method overload should either have the CultureInfo parameter as the first argument or as the last argument. If we assume the CultureInfo is going to be the last parameter of the overload we pick, then we need to compare the parameters of the current invoked method to the parameters of the overload method we recommend from 0 index. If the CultureInfo is going to the first parameter then we need to compare the parameters with a offset of 1 in the overload method to make sure the rest of the parameters in the method are the same.